src/main.rs

```diff
@@ -50,6 +56,11 @@ impl<'a, 'b> ggez::event::EventHandler for State<'a, 'b> {
                     .with(Faction::Player)
                     .with(Shooter { seconds_per_attack: 1.0, cooldown: 0.0, attack_radius: 100.0 })
                     .build();
+
+                // Spend resources!
+                // TODO: Make the cost tunable in data somehow.
```

Comment:
Agreed! Maybe a "Cost" component that can be initialized on creation. That will let us do silly things like make a monster tower that will spawn monsters when clicked but only at a cost.

Reply:
Yeah, I think a component could work well :) Let's have a design discussion!